integreat_cms/core/management/commands/copy_pois_events_and_contacts.py

```diff
@@ -0,0 +1,207 @@
+
+if TYPE_CHECKING:
+    from typing import Any
+
+    from django.core.management.base import CommandParser
+    from django.db.models import FileField
+
+    from integreat_cms.cms.models import EventTranslation, POITranslation
+
+    from .users.user import User
+
+
+logger = logging.getLogger(__name__)
+
+
+def make_hardlink(file_field: FileField, region_id: int) -> FileField:
+    base = file_field.storage.base_location
+    orig_path = file_field.name
+    new_path = re.sub(r"^regions/[0-9]+/", f"regions/{region_id}/", orig_path)
+    # make the directory
+    os.makedirs(os.path.dirname(os.path.join(base, new_path)), exist_ok=True)
+    # create the hard link
+    try:
+        os.link(os.path.join(base, orig_path), os.path.join(base, new_path))
+    except FileExistsError as e:
+        if (
+            os.stat(os.path.join(base, orig_path)).st_ino
+            == os.stat(os.path.join(base, new_path)).st_ino
+        ):
+            logger.info(
+                "Hard link already exists between %r and %r",
+                new_path,
+                orig_path,
+            )
+        else:
+            raise FileExistsError("Failed to create hard link") from e
+    # modify the file information
+    file_field.name = new_path
+    return file_field
+
+
+def copy_icon(region: Region, obj: POI | Event) -> None:
+    if obj.icon and obj.icon.region is not None:
+        # Theoretically, there might be orphaned media files
+        obj.icon.pk = None
+        obj.icon.region = region
+        try:
+            obj.icon.file = make_hardlink(obj.icon.file, region.pk)
+        except FileNotFoundError:
+            logger.info("Could not copy icon for %r: FileNotFound", obj)
+        obj.icon.save()
+
+
+def copy_translations(
+    region: Region,
+    translations: list[POITranslation] | list[EventTranslation],
+    obj: POI | Event,
+    related_name: str,
+) -> None:
+    for tr in translations:
+        if tr.language in region.languages:
+            # Ensure partial attempts don't make everything more difficult
+            try:
+                dup = obj.translations.get(language=tr.language, version=tr.version)
+            except obj.translations.model.DoesNotExist:
+                pass
+            else:
+                logger.info(
+                    "Not saving translation, already exists in %r: %r",
+                    tr.language,
+                    dup,
+                )
+                continue
+            tr.pk = None
+            setattr(tr, related_name, obj)
+            # Don't update the timestamp
+            tr.save(update_timestamp=False)
+            logger.info(
+                "Saved translation to target region in %r: %r",
+                tr.language,
+                tr,
+            )
+        else:
+            logger.info(
+                "Not saving translation in language not in target region (%r): %r",
+                tr.language,
+                tr,
+            )
```

Comment:
Why not make this a continue statement? It might make this a bit more readable
```suggestion
        if tr.language not in region.languages:
            logger.info(
                "Not saving translation in language not in target region (%r): %r",
                tr.language,
                tr,
            )
            continue

        # Ensure partial attempts don't make everything more difficult
        try:
            dup = obj.translations.get(language=tr.language, version=tr.version)
        except obj.translations.model.DoesNotExist:
            pass
        else:
            logger.info(
                "Not saving translation, already exists in %r: %r",
                tr.language,
                dup,
            )
            continue
        tr.pk = None
        setattr(tr, related_name, obj)
        # Don't update the timestamp
        tr.save(update_timestamp=False)
        logger.info(
            "Saved translation to target region in %r: %r",
            tr.language,
            tr,
        )
```